### PR TITLE
TSK-1871: Implement new Frontend structure

### PR DESCRIPTION
**Pull request by knht**
Co-authored-by: Chi Nguyen <c.nguyen.prog@gmail.com>

Co-authored-by: Mustapha Zorgati <15628173+mustaphazorgati@users.noreply.github.com>

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: web/angular.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
         "build": {
           "builder": "@angular-builders/custom-webpack:browser",
           "options": {
+            "allowedCommonJsDependencies": ["chart.js"],
```
Comment (knht) [inline web/angular.json:14]: Possible purge of `chart.js` dependency in favor of a native implementation during later rewriting process

File: web/src/app/v2/task/components/task-container/task-container.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { TaskContainerComponent } from './task-container.component';
+
+describe.skip('TaskContainerComponent', () => {
```
Comment (knht) [inline web/src/app/v2/task/components/task-container/task-container.component.spec.ts:5]: Revert test skipping once testing for said component becomes relevant again.

File: web/src/app/v2/task/components/task-details-container/task-details-container.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { TaskDetailsContainerComponent } from './task-details-container.component';
+
+describe.skip('TaskDetailsContainerComponent', () => {
```
Comment (knht) [inline web/src/app/v2/task/components/task-details-container/task-details-container.component.spec.ts:5]: Revert test skipping once testing for said component becomes relevant again.

File: web/src/app/v2/task/components/task-details-container/task-details-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+@Component({
+  selector: 'taskana-task-details-container',
+  templateUrl: './task-details-container.component.html',
+  styleUrls: ['./task-details-container.component.scss']
+})
+export class TaskDetailsContainerComponent implements OnInit {
+  constructor(private route: ActivatedRoute, private taskFacade: TaskFacadeService) {}
+
+  ngOnInit(): void {
+    //TODO add takeuntil destroy, can use https://github.com/ngneat/until-destroy
```
Comment (knht) [inline web/src/app/v2/task/components/task-details-container/task-details-container.component.ts:15]: Implement `until-destroy` library in favor of manual component lifecycle destroying

File: web/src/app/v2/task/components/task-list/task-list.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { TaskListComponent } from './task-list.component';
+
+describe.skip('TaskListComponent', () => {
```
Comment (knht) [inline web/src/app/v2/task/components/task-list/task-list.component.spec.ts:5]: Revert test skipping once testing for said component becomes relevant again.

File: web/src/app/v2/task/components/task-list/task-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  @Select(TaskSelector.selectedTask) selectedTask$: Observable<Task | null>;
+
+  constructor(private taskFacade: TaskFacadeService) {}
+
+  ngOnInit(): void {}
+
+  selectTask(taskId: string): void {
+    if (this.isTaskSelected(taskId)) {
+      /**
+       * @TODO Add deselectTask to facade
```
Comment (knht) [inline web/src/app/v2/task/components/task-list/task-list.component.ts:26]: **Important** Deselecting needs to be implemented in the facade and store side

File: web/src/app/v2/task/services/task-facade.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { TaskFacadeService } from './task-facade.service';
+
+describe.skip('TaskFacadeService', () => {
```
Comment (knht) [inline web/src/app/v2/task/services/task-facade.service.spec.ts:5]: Revert test skipping once testing for said component becomes relevant again.

File: web/src/app/v2/task/services/task-facade.service.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    return this.store.selectSnapshot(TaskSelector.selectedTask);
+  }
+
+  getTasks(): void {
+    this.store.dispatch(new GetTasks());
+  }
+
+  // distinguish between select a task (what happens to UI?) and the action of getting information of a task from a taskID
+  selectTask(taskId: string): void {
+    this.store.dispatch(new Navigate([`/taskana/workplace/tasks/taskdetail/${taskId}`]));
```
Comment (knht) [inline web/src/app/v2/task/services/task-facade.service.ts:25]: Make NGXS `Navigate` call dynamic by extracting the path to a constants file exporting functions for generating dynamic routes.

File: web/src/app/v2/task/services/task.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { TaskService } from './task.service';
+
+describe.skip('TaskService', () => {
```
Comment (knht) [inline web/src/app/v2/task/services/task.service.spec.ts:5]: Revert test skipping once testing for said component becomes relevant again.

File: web/src/app/v2/task/store/task.selector.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { Selector } from '@ngxs/store';
+import { Task, TaskSummary } from '@task/models/task';
+import { TaskState, TaskStateModel } from './task.state';
+
+@Injectable()
+export class TaskSelector {
+  @Selector([TaskState])
+  static tasks(state: TaskStateModel): TaskSummary[] {
+    return state.pagedTask?.tasks || []; /* @TODO Initialize page tasks with an initial value */
```
Comment (mustaphazorgati) [inline web/src/app/v2/task/store/task.selector.ts:10]:
```suggestion
    return state.pagedTask.tasks

File: web/src/app/v2/task/store/task.state.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+@State<TaskStateModel>({
+  name: 'task',
+  defaults
+})
+@Injectable()
+export class TaskState {
+  constructor(private taskService: TaskService) {}
+
+  @Action(GetTasks)
+  getTasks(ctx: StateContext<TaskStateModel>) {
```
Comment (mustaphazorgati) [inline web/src/app/v2/task/store/task.state.ts:28]: Typescript: return type missing

File: web/src/app/v2/task/store/task.state.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    return this.taskService.getTasks().pipe(
+      take(1),
+      tap((pagedTask) => {
+        ctx.patchState({ pagedTask });
+      })
+    );
+  }
+
+  @Action(GetTask)
+  getTask(ctx: StateContext<TaskStateModel>, { taskId }: GetTask) {
```
Comment (mustaphazorgati) [inline web/src/app/v2/task/store/task.state.ts:38]: Typescript: return type missing

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

**Pull request merged**